### PR TITLE
added median CI and messaging

### LINKED ISSUE #995: Median CI by default for continuous variable in add_ci()
By default, `tbl_summary()` displays the median for continuous variables. Therefore, it would be more consistent that `add_ci()` computes by default the confidence interval of the median for continuous variables.

It could be easily done with `wilcox.test(conf.int = TRUE)`.

Therefore I suggest:

- [ ] add a `wilcox.test` method to `add_ci()`
- [ ] by default, use `wilcox.test` for continuous variables
- [ ] if `theme_gtsummary_mean_sd()` is applied, change default to `t.test`

### LINKED ISSUE #998: update tbl_regression documentation
The note section on N in the help file of tbl_regression is not needed, since N column is not a default output anymore. Can you update? Thanks.

### PR DESCRIPTION
**Pull request by ddsjoberg**
**What changes are proposed in this pull request?**

**If there is an GitHub issue associated with this pull request, please provide link.**
closes #995
closes #998

--------------------------------------------------------------------------------

Checklist for PR reviewer

- [ ] PR branch has pulled the most recent updates from master branch. Ensure the pull request branch and your local version match and both have the latest updates from the master branch.
- [ ] If an update was made to `tbl_summary()`, was the same change implemented for `tbl_svysummary()`?
- [ ] If a new function was added, function included in `_pkgdown.yml`
- [ ] If a bug was fixed, a unit test was added for the bug check
- [ ] Run `pkgdown::build_site()`. Check the R console for errors, and review the rendered website.
- [ ] Code coverage is suitable for any new functions/features. Review coverage with `covr::report()`. Before you run, set `Sys.setenv(NOT_CRAN="true")` and begin in a fresh R session without any packages loaded. 
- [ ] R CMD Check runs without errors, warnings, and notes
- [ ] `usethis::use_spell_check()` runs with no spelling errors in documentation

When the branch is ready to be merged into master:
- [ ] Update NEWS.md with the changes from this pull request under the heading "`# gtsummary (development version)`". If there is an issue associated with the pull request, reference it in parantheses at the end update (see `NEWS.md` for examples).
- [ ] Increment the version number using `usethis::use_version(which = "dev")` 
- [ ] Run `codemetar::write_codemeta()`
- [ ] Run `usethis::use_spell_check()` again
- [ ] Approve Pull Request
- [ ] Merge the PR. Please use "Squash and merge".

### CODE REVIEW

**Comment by ddsjoberg** [comment]
POTENTIAL ISSUE!!
I updated `broom.helpers::.formula_list_to_named_list()` recently....I may have added a bug, where we return an error when a selector doesn't select any variables. Previously, it was fine to use `all_continuous()` in a context when there were no continuous variables in the table.

**Comment by ddsjoberg** [comment]
Hmm, maybe that was not the issue...need to look into this further...

``` r
library(gtsummary)
trial %>%
  select(grade, response, trt) %>%
  tbl_summary(missing = "no", by = trt) %>%
  add_p(all_continuous() ~ "t.test") %>%
  as_kable()
```

| **Characteristic** | **Drug A**, N = 98 | **Drug B**, N = 102 | **p-value** |
|:-------------------|:-------------------|:--------------------|:------------|
| Grade              |                    |                     | 0.9         |
| I                  | 35 (36%)           | 33 (32%)            |             |
| II                 | 32 (33%)           | 36 (35%)            |             |
| III                | 31 (32%)           | 33 (32%)            |             |
| Tumor Response     | 28 (29%)           | 33 (34%)            | 0.5         |

<sup>Created on 2021-10-04 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

File: R/add_ci.R [hunk 1]
```diff
@@ -242,12 +245,29 @@ single_ci <- function(variable, by, tbl, method, conf.level,
...
+      cli::cli_alert_danger()
+    }
     df_ci <-
       stats::t.test(data[[variable]], conf.level = conf.level) %>%
       broom::tidy()
   }
+  else if (method %in% "wilcox.test") {
+    if (!"median" %in%
+        names(tbl$meta_data[tbl$meta_data$variable %in% variable, ]$df_stats[[1]])) {
+      paste("{.code add_ci()} added psuedo-median CI for {.val {variable}};",
```
Comment (larmarange) [inline R/add_ci.R:263]: typo

**Comment by larmarange** [comment]
> Hmm, maybe that was not the issue...need to look into this further...

Let me know if any correction is required in `broom.helpers`

**Comment by ddsjoberg** [comment]
oy, nevermind! I think it was just a typo in the example code 😬 😬 😬 
@larmarange

File: R/add_ci.R [hunk 2]
```diff
@@ -23,7 +23,8 @@
 #' `prop.test(correct = c(TRUE, FALSE))`.
 #' The default method, `"wilson"`, includes the Yates continuity correction.
 #' Methods `c("exact", "asymptotic")` are calculated with `Hmisc::binconf(method=)`.
-#' Confidence intervals for means are calculated using `t.test()`.
+#' Confidence intervals for means are calculated using `t.test()` and
+#' `wilcox.test()` for psuedo-medians.
```
[the remark this answers is not in the document]
Comment (ddsjoberg) [reply R/add_ci.R:27]: thank you!

**Comment by ddsjoberg** [comment]
@larmarange does this include the functionality/messaging you asked about?

**Review by larmarange — APPROVED** [review]

**Comment by larmarange** [comment]
> @larmarange does this include the functionality/messaging you asked about?

It's good for me. Thanks!!

**Pull request merged**